src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -0,0 +1,197 @@
+     */
+    private String getMessage(MessageContext messageContext) throws AxisFault {
+        Axis2MessageContext axisMsgContext = (Axis2MessageContext) messageContext;
+        org.apache.axis2.context.MessageContext msgContext = axisMsgContext.getAxis2MessageContext();
+        return formatMessage(msgContext);
+    }
+
+    /**
+     * Format the messages when the messages are sent to the kafka broker
+     */
+    private static String formatMessage(org.apache.axis2.context.MessageContext messageContext) throws AxisFault {
+        OMOutputFormat format = BaseUtils.getOMOutputFormat(messageContext);
+        MessageFormatter messageFormatter = MessageProcessorSelector.getMessageFormatter(messageContext);
+        StringWriter stringWriter = new StringWriter();
+        OutputStream out = new WriterOutputStream(stringWriter, format.getCharSetEncoding());
+        try {
+            messageFormatter.writeTo(messageContext, format, out, true);
+        } catch (IOException e) {
+            throw new AxisFault("The Error occurs while formatting the message", e);
+        } finally {
+            try {
+                out.close();
+            } catch (Exception e) {
+                throw new AxisFault("The Error occurs while closing the output stream", e);
+            }
+        }
+
+        return stringWriter.toString();
+    }
+
+    /**
+     * Send the messages to the kafka broker with topic and the key that is optional.
+     *
+     * @param producer    The instance of the Kafka producer.
+     * @param topic       The topic to send the message.
+     * @param partitionNo The partition Number of the broker.
+     * @param key         The key.
+     * @param message     The message that send to kafka broker.
+     */
+    private void send(KafkaProducer<String, String> producer, String topic, String partitionNo, String key,
+            String message) {
+        if (partitionNo == null) {
+            producer.send(new ProducerRecord<String, String>(topic, message));
+            producer.flush();
+        } else {
+            producer.send(new ProducerRecord<>(topic, Integer.parseInt(partitionNo), key, message));
+            producer.flush();
+        }
+    }
+
+    /**
+     * Send the messages with connection pool.
+     *
+     * @param messageContext The message context.
+     * @param topic          The topic.
+     * @param partitionNo    The partition Number of the broker.
+     * @param key            The key.
+     * @param message        The message.
+     * @throws ConnectException The Exception while create the connection from the Connection pool.
+     */
+    private void sendWithPool(MessageContext messageContext, String topic, String partitionNo, String key,
+            String message) throws ConnectException {
+
+        KafkaProducer<String, String> producer = KafkaConnectionPool.getConnectionFromPool();
+        if (producer == null) {
```

Comment:
Here we check the existence of the connection and if there is no connection we need to create the connection pool and get the kafka connection from the pool. Then only we send the message using that connection.